code/game/machinery/kitchen/cooking_machines/_appliance.dm

```diff
@@ -0,0 +1,652 @@
+		return span("soghun","<b>It is done !</b>")
+	if (progress < CI.overcook_mult)
+		return span("warning","It looks overcooked, get it out!")
+	else
+		return span("danger","It is burning!!")
+
+/obj/machinery/appliance/update_icon()
+	if (!stat && cooking_objs.len)
+		icon_state = on_icon
+
+	else
+		icon_state = off_icon
+
+/obj/machinery/appliance/verb/toggle_power()
+	set src in view()
+	set name = "Toggle Power"
+	set category = "Object"
+
+	if (!isliving(usr))
+		usr << "Ghosts aren't allowed to toggle power switches"
+		return
+
+	if (!Adjacent(usr))
+		if (!istype(usr, /mob/living/silicon))
+			usr << "You can't reach the power switch from there, get closer!"
+			return
+
+	if (stat & POWEROFF)//Its turned off
+		stat &= ~POWEROFF
+		if (usr)
+			usr.visible_message("[usr] turns the [src] on", "You turn on the [src]")
+	else //Its on, turn it off
+		stat |= POWEROFF
+		if (usr)
+			usr.visible_message("[usr] turns the [src] off", "You turn off the [src]")
+	playsound(src, 'sound/machines/click.ogg', 40, 1)
+	update_icon()
+
+/obj/machinery/appliance/proc/choose_output()
+	set src in view()
+	set name = "Choose output"
+	set category = "Object"
+
+	if(output_options.len)
+
+		var/choice = input("What specific food do you wish to make with \the [src]?") as null|anything in output_options+"Default"
+		if(!choice)
+			return
+		if(choice == "Default")
+			selected_option = null
+			usr << "<span class='notice'>You decide not to make anything specific with \the [src].</span>"
+		else
+			selected_option = choice
+			usr << "<span class='notice'>You prepare \the [src] to make \a [selected_option] with the next thing you put in. Try putting several ingredients in a container!</span>"
+
+//Handles all validity checking and error messages for inserting things
+/obj/machinery/appliance/proc/can_insert(var/obj/item/I, var/mob/user)
+	if(!dropsafety(I))
+		return 0
+
+	// We are trying to cook a grabbed mob.
+	var/obj/item/weapon/grab/G = I
+	if(istype(G))
+
+		if(!can_cook_mobs)
+			user << "<span class='warning'>That's not going to fit.</span>"
+			return 0
+
+		if(!isliving(G.affecting))
+			user << "<span class='warning'>You can't cook that.</span>"
+			return 0
+
+		return 2
+
+
+	if (!has_space(I))
+		user << "<span class='warning'>There's no room in [src] for that!</span>"
+		return 0
+
+
+	if (container_type && istype(I, container_type))
+		return 1
+
+	// We're trying to cook something else. Check if it's valid.
+	var/obj/item/weapon/reagent_containers/food/snacks/check = I
+	if(istype(check) && islist(check.cooked) && (cook_type in check.cooked))
+		user << "<span class='warning'>\The [check] has already been [cook_type].</span>"
+		return 0
+	else if(istype(check, /obj/item/weapon/reagent_containers/glass))
+		user << "<span class='warning'>That would probably break [src].</span>"
+		return 0
+	else if(istype(check, /obj/item/weapon/disk/nuclear))
+		user << "Central Command would kill you if you [cook_type] that."
```

Comment:
ehhh
this is pre-existing code, it was put in there i imagine because the first thing everyone wants to do is deepfry the nuke disk. i didnt really touch this except to move it around.

Its hard to tell what i touched since i renamed files, so it looks like everything is written from scratch